### PR TITLE
[INLONG-8611][Manager] Add full link trace based on Opentelemetry

### LINKED ISSUE #8611: [Feature][Manager][Agent][DataProxy] Support full link tracking to improve the observability of the project
### Description

Inlong lacks observability, which is not conducive to link tracking and error location.

### InLong Component

InLong Manager, InLong Agent, InLong DataProxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ZhaoNiuniu**
### Prepare a Pull Request

- Title Example: [INLONG-8611][Manager] Add full link trace.

- Fixes #8611 

### Motivation

*Inlong lacks observability, which is not conducive to link tracking and error location.*

### Modifications
Oentelemetry java agent is introduced for link embedding, collection and reporting, and span log is added to realize keyword search. It mainly includes the following functions:
- Opentelemetry java agent has automatically collected some fixed attributes for each span. In order to better meet the needs of inlong's link tracking, we have added some special attributes for span. Such as inlongGroupId, inlongstream, clusterName, clusterTag. Enter "key = value", for example, "inlongGroupId=test-group", and relevant links will be found. 
image

- Whether each response succeeds or fails, the trace id of its link can be found in the response header.
image

- An event log is added to the span of the link portal, which shows the request and response contents of the current request.
image

- In the output log, if the log belongs to a link, trace id and span id will be printed when the log is printed. 
image

### CODE REVIEW

File: inlong-agent/bin/agent-env.sh [hunk 1]
```diff
@@ -61,3 +61,19 @@ if [[ $JMX_ENABLED == 1 ]]; then
...
+    wget -O "$JAVA_AGENT" "$DOWNLOAD_URL"
+fi
+
+# Opentelemetry startup parameter configuration
+export OTEL_SERVICE_NAME=inlong_agent
+export OTEL_TRACES_EXPORTER=otlp
+export OTEL_METRICS_EXPORTER=otlp
+export OTEL_LOGS_EXPORTER=otlp
+export OTEL_INSTRUMENTATION_PULSAR_ENABLED=false
+export OTEL_EXPORTER_OTLP_ENDPOINT="http://otel-collect-proxy.zhiyan.tencent-cloud.net:4317"
```
Comment (vernedeng) [inline inlong-agent/bin/agent-env.sh:78]: Do not put the internal address to the community

**Review by vernedeng — CHANGES_REQUESTED** [review]

File: inlong-agent/bin/agent-env.sh [hunk 2]
```diff
@@ -61,3 +61,19 @@ if [[ $JMX_ENABLED == 1 ]]; then
 else
   export AGENT_ARGS="$AGENT_JVM_ARGS -cp $CLASSPATH -Dagent.home=$BASE_DIR"
 fi
+
+# Opentelemetry java agent path
+JAVA_AGENT="${BASE_DIR}/bin/opentelemetry-javaagent.jar"
+DOWNLOAD_URL="https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/download/v1.28.0/opentelemetry-javaagent.jar"
```
Comment (vernedeng) [inline inlong-agent/bin/agent-env.sh:67]: Make the version of open-telemetry configurable

**Comment by dockerzhang** [comment]
Please rebase from the master to solve the timeout workflow issue.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Override
+    public boolean isReady() {
+        if (response == null) {
+            return false;
+        }
+        try {
+            return response.getOutputStream().isReady();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java:48]:
```suggestion
        LOG.err("got exception when check if  the output stream is ready", e)
```

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+        return false;
+    }
+
+    @Override
+    public void setWriteListener(WriteListener listener) {
+        if (response != null) {
+            try {
+                ((CoyoteOutputStream) response.getOutputStream()).setWriteListener(listener);
+            } catch (IOException e) {
+                e.printStackTrace();
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java:59]: ditto

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+            }
+        }
+    }
+
+    @Override
+    public void write(int b) throws IOException {
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java:65]: **_b_** is not a readable name

File: pom.xml [hunk 6]
```diff
@@ -205,6 +205,14 @@
 
     <dependencyManagement>
         <dependencies>
+            <!--opentelemetry-->
+            <dependency>
+                <groupId>io.opentelemetry</groupId>
+                <artifactId>opentelemetry-bom</artifactId>
+                <version>1.24.0</version>
```
Comment (vernedeng) [inline pom.xml:212]: Please define the version as a reference

Comment (ZhaoNiuniu) [reply pom.xml:212]: The parameter name is consistent with the rewritten function.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java (hunk 5 above)
Comment (ZhaoNiuniu) [reply inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java:65]: The parameter name is consistent with the rewritten function.

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.catalina.connector.CoyoteOutputStream;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.OutputStream;
```
Comment (vernedeng) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/trace/WrapperOutputStream.java:28]: Need one blank line

File: inlong-agent/bin/agent-env.sh [hunk 8]
```diff
@@ -61,3 +66,6 @@ if [[ $JMX_ENABLED == 1 ]]; then
 else
   export AGENT_ARGS="$AGENT_JVM_ARGS -cp $CLASSPATH -Dagent.home=$BASE_DIR"
 fi
+
+# Opentelemetry java agent path
+JAVA_AGENT="${BASE_DIR}/lib/opentelemetry-javaagent-${OTEL_VERSION}.jar"
```
Comment (vernedeng) [inline inlong-agent/bin/agent-env.sh:71]: Please rename **_JAVA_AGENT_**  to something like **_OTEL_AGENT_**

**Review by vernedeng — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Pull request merged**